Newly created Go extensions using standard workflow template fail due to older `cli/gh-extension-precompile` action and Android builds

### Describe the bug

With the `v2` major release of `cli/gh-extension-precompile` addressing build issues with more recent Go releases on Android in https://github.com/cli/gh-extension-precompile/pull/56, the release workflow templates used by `gh ext create` should be updated as the `v1` version causes Go extensions to break.

```shell
$ gh --version
gh version 2.62.0 (2024-11-14)
https://github.com/cli/cli/releases/tag/v2.62.0
```

https://github.com/cli/cli/blob/a50fc7079e574a68284e19ea7eaee2f8303e79f0/pkg/cmd/extension/ext_tmpls/goBinWorkflow.yml#L11-L19

https://github.com/cli/cli/blob/a50fc7079e574a68284e19ea7eaee2f8303e79f0/pkg/cmd/extension/ext_tmpls/otherBinWorkflow.yml#L9-L16

### Steps to reproduce the behavior

1. Create standard Go extension

   ```shell
   gh ext create --precompiled=go foobar
   ```

1. Create repository based on local repository

   ```shell
   cd gh-foobar
   gh repo create gh-acceptance-testing/gh-foorbar --private --push --source .
   ```

1. Create release

   ```shell
   gh release create v0.0.1 --title v0.0.1 --generate-notes
   ```

1. Confirm release workflow fails due to Android CGO

   ```shell
   gh run view "$(gh run list --json databaseId --jq '.[0].databaseId')" --log-failed
   ```

   resulting in:
    
   ```shell
   android/amd64 requires external (cgo) linking, but cgo is not enabled
   Error: Process completed with exit code 1.
   ```
    
### Expected vs actual behavior

I expect newly created extensions to be releasable from an unmodified standard release workflow with current Go releases.

--- a/pkg/cmd/extension/ext_tmpls/goBinWorkflow.yml
+++ b/pkg/cmd/extension/ext_tmpls/goBinWorkflow.yml
@@ -13,7 +13,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
-      - uses: cli/gh-extension-precompile@v1
+      - uses: cli/gh-extension-precompile@v2
         with:
           generate_attestations: true
           go_version_file: go.mod

--- a/pkg/cmd/extension/ext_tmpls/otherBinWorkflow.yml
+++ b/pkg/cmd/extension/ext_tmpls/otherBinWorkflow.yml
@@ -11,6 +11,6 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
-      - uses: cli/gh-extension-precompile@v1
+      - uses: cli/gh-extension-precompile@v2
         with:
           build_script_override: "script/build.sh"

